### ISSUE TITLE
False positive `R1733(unnecessary-dict-index-lookup)` when deleting item of dict

### ISSUE THREAD

**Issue by starflows** [PyCQA/pylint#5458]
### Bug description

`d.py`

```python
# pylint: disable=C0114
d = {'key': 'value'}

for k, v in d.items():
    if v == 'value':
        del d[k]
        break

print(d)
```

### Command used

```shell
pylint d.py
```

### Pylint output

```shell
************* Module d
d.py:6:12: R1733: Unnecessary dictionary index lookup, use 'v' instead (unnecessary-dict-index-lookup)

-----------------------------------
Your code has been rated at 8.33/10
```

### Expected behavior

no `R1733` when using `del`

the proposed change (use `v` instead) does not work:

```python
# pylint: disable=C0114
d = {'key': 'value'}

for k, v in d.items():
    if v == 'value':
        del v  # does not remove item from dict
        break

print(d)
```

### Pylint version

```shell
pylint 2.12.1
astroid 2.9.0
Python 3.9.9 (main, Nov 20 2021, 21:30:06) 
[GCC 11.1.0]
```

**Comment by DanielNoord** [comment]
This is a duplicate of #4716. A fix has been made in #5344 which will be released with `pylint` `2.13`!
